### PR TITLE
read podManifestPath from kubeadm file

### PR DESCRIPTION
**Pull request by adamzhoul**
#### What type of PR is this?

> /kind enhancement

#### Which issue(s) this PR fixes:

Fixes https://github.com/openyurtio/openyurt/issues/460

#### Special notes for your reviewer:

/assign @Peeknut 

 removed `pod-manifest-path`  for both `yurtctl convert` and `yurtctl revert`
 

#### Does this PR introduce a user-facing change?

```release-note
   rm --pod-manifest-path

```

### CODE REVIEW

File: pkg/yurtctl/cmd/convert/edgenode.go [hunk 1]
```diff
@@ -155,6 +141,12 @@ func (c *ConvertEdgeNodeOptions) Complete(flags *pflag.FlagSet) error {
 	}
 	c.KubeadmConfPath = kubeadmConfPath
 
+	podManifestPath, err := enutil.GetPodManifestPath(c.KubeadmConfPath)
+	if err != nil {
+		return err
+	}
+	c.PodMainfestPath = podManifestPath
```
Comment (Peeknut) [inline pkg/yurtctl/cmd/convert/edgenode.go:148]: As `podManifestPath` only use in func `SetupYurthub()`, how about remove `c.PodManifestPath`?

File: pkg/yurtctl/cmd/revert/edgenode.go [hunk 2]
```diff
@@ -123,6 +109,12 @@ func (r *RevertEdgeNodeOptions) Complete(flags *pflag.FlagSet) (err error) {
 	}
 	r.KubeadmConfPath = kubeadmConfPath
 
+	podManifestPath, err := enutil.GetPodManifestPath(r.KubeadmConfPath)
+	if err != nil {
+		return err
+	}
+	r.PodMainfestPath = podManifestPath
```
Comment (Peeknut) [inline pkg/yurtctl/cmd/revert/edgenode.go:116]: As `podManifestPath` use in func `RemoveYurthub()`, how about remove c.PodManifestPath?

File: pkg/yurtctl/cmd/revert/revert.go [hunk 3]
```diff
@@ -32,6 +32,7 @@ import (
 	"github.com/openyurtio/openyurt/pkg/projectinfo"
 	"github.com/openyurtio/openyurt/pkg/yurtctl/constants"
 	"github.com/openyurtio/openyurt/pkg/yurtctl/lock"
+	enutil "github.com/openyurtio/openyurt/pkg/yurtctl/util/edgenode"
```
Comment (Peeknut) [inline pkg/yurtctl/cmd/revert/revert.go:35]: Better to move it to make it comply with the previous rules.

Comment (adamzhoul) [reply pkg/yurtctl/cmd/revert/revert.go:35]: sorry,  which rule do you mean?

the go pkg，third pkg, self pkg. 
following this rule,  `enutil "github.com/openyurtio/openyurt/pkg/yurtctl/util/edgenode"`  shall be placed here 
after `yurtctl/lock`

file `pkg/yurtctl/cmd/convert/convert.go` is not followed, but what about this ?

File: pkg/yurtctl/cmd/revert/edgenode.go (hunk 2 above)
Comment (adamzhoul) [reply pkg/yurtctl/cmd/revert/edgenode.go]: updated

File: pkg/yurtctl/cmd/convert/edgenode.go (hunk 1 above)
Comment (adamzhoul) [reply pkg/yurtctl/cmd/convert/edgenode.go]: updated

File: pkg/yurtctl/cmd/revert/revert.go (hunk 3 above)
Comment (Peeknut) [reply pkg/yurtctl/cmd/revert/revert.go:35]: sorry, it's right here.

File: pkg/yurtctl/cmd/convert/edgenode.go [hunk 4]
```diff
@@ -403,8 +387,8 @@ func (c *ConvertEdgeNodeOptions) ResetKubelet() error {
 	return nil
 }
 
-func (c *ConvertEdgeNodeOptions) getYurthubYaml() string {
-	return filepath.Join(c.PodMainfestPath, enutil.YurthubYamlName)
+func (c *ConvertEdgeNodeOptions) getYurthubYaml(podManifestPath string) string {
```
Comment (Peeknut) [inline pkg/yurtctl/cmd/convert/edgenode.go:390]: How about remove func `getYurthubYaml()` as this func is not related to `ConvertEdgeNodeOptions`.

File: pkg/yurtctl/cmd/revert/edgenode.go [hunk 5]
```diff
@@ -315,8 +303,8 @@ func (r *RevertEdgeNodeOptions) getKubeletSvcBackup() string {
 	return fmt.Sprintf(enutil.KubeletSvcBackup, r.KubeadmConfPath)
 }
 
-func (r *RevertEdgeNodeOptions) getYurthubYaml() string {
-	return filepath.Join(r.PodMainfestPath, enutil.YurthubYamlName)
+func (r *RevertEdgeNodeOptions) getYurthubYaml(podManifestPath string) string {
```
Comment (Peeknut) [inline pkg/yurtctl/cmd/revert/edgenode.go:306]: How about remove func `getYurthubYaml()` as this func is not related to `RevertEdgeNodeOptions`?

File: pkg/yurtctl/cmd/convert/convert.go [hunk 6]
```diff
@@ -22,6 +22,7 @@ import (
 	"strings"
 	"time"
 
+	enutil "github.com/openyurtio/openyurt/pkg/yurtctl/util/edgenode"
```
[the remark this answers is not in the document]
Comment (adamzhoul) [reply pkg/yurtctl/cmd/convert/convert.go]: updated

File: pkg/yurtctl/cmd/convert/edgenode.go (hunk 4 above)
Comment (adamzhoul) [reply pkg/yurtctl/cmd/convert/edgenode.go:390]: defining the YAML file path is the responsibility of convert.
so are the functions `getYurthubKubeletConf` `getKubeletSvcBackup`

it may not be related to object `ConvertEdgeNodeOptions`, so keep it as a pure function?

keep it can still make some sense. what do you think?

File: pkg/yurtctl/cmd/revert/edgenode.go (hunk 5 above)
Comment (adamzhoul) [reply pkg/yurtctl/cmd/revert/edgenode.go:306]: same as `ConvertEdgeNodeOptions`

File: pkg/yurtctl/cmd/convert/edgenode.go (hunk 4 above)
Comment (Peeknut) [reply pkg/yurtctl/cmd/convert/edgenode.go:390]: Agree, can change it to a pure func

Comment (adamzhoul) [reply pkg/yurtctl/cmd/convert/edgenode.go:390]: what about `getYurthubKubeletConf` `getKubeletSvcBackup`
change too ?

Comment (Peeknut) [reply pkg/yurtctl/cmd/convert/edgenode.go:390]: Ok. And func `getYurthubConf()` in `pkg/yurtctl/cmd/revert/edgenode.go` can change together.

Comment (adamzhoul) [reply pkg/yurtctl/cmd/convert/edgenode.go:390]: after a second thought.

the root problem may be : 
`
      func (r *RevertEdgeNodeOptions) RunRevertEdgeNode() (err error) {
`
this core function related to `RevertEdgeNodeOptions`, so in the function, we use r.xx everywhere 

the better way may be：
1.  pure function   ` func RunRevertEdgeNode(o RevertEdgeNodeOptions)`
2. call it in 
```
	r := NewRevertEdgeNodeOptions()
        ... 
        Run: func(cmd *cobra.Command, _ []string) {
                         ...
			RunRevertEdgeNode(r)
		},
```

what do you think ？

File: pkg/yurtctl/util/edgenode/util_test.go [hunk 7]
```diff
@@ -0,0 +1,40 @@
+package edgenode
```
Comment (rambohe-ch) [inline pkg/yurtctl/util/edgenode/util_test.go:1]: please add license header.

Comment (adamzhoul) [reply pkg/yurtctl/util/edgenode/util_test.go:17]: done

Comment (adamzhoul) [reply pkg/yurtctl/util/edgenode/util_test.go:17]: do we have a hack script to add automatically？
and add a ci for this?

File: pkg/yurtctl/cmd/convert/edgenode.go (hunk 4 above)
Comment (Peeknut) [reply pkg/yurtctl/cmd/convert/edgenode.go:390]: Both implementation methods are OK, but I prefer to keep the original method because it doesn't affect anything.
Just change these functions to public and pure functions so that convert and revert can be called at the same time.

File: pkg/yurtctl/cmd/revert/edgenode.go (hunk 5 above)
Comment (adamzhoul) [reply pkg/yurtctl/cmd/revert/edgenode.go]: done

File: pkg/yurtctl/cmd/convert/edgenode.go (hunk 4 above)
Comment (adamzhoul) [reply pkg/yurtctl/cmd/convert/edgenode.go]: done.
but keep them as privite.

File: pkg/yurtctl/util/edgenode/util_test.go (hunk 7 above)
Comment (rambohe-ch) [reply pkg/yurtctl/util/edgenode/util_test.go:17]: > do we have a hack script to add automatically？
> and add a ci for this?

hack script is not provided now. and i will make an issue about it.

**Comment by rambohe-ch** [comment]
@adamzhoul Thank you for your contribution, I want to remind you again. We have sent github message to you for inviting you as community member,  please check the issue: https://github.com/openyurtio/community/issues/25
